### ISSUE TITLE
Add support for clang-format configuration file

### ISSUE THREAD

**Issue by ferrarimarco** [github/super-linter#1829]
**Is your feature request related to a problem? Please describe.**

clang-format supports passing a [configuration file](https://clang.llvm.org/docs/ClangFormatStyleOptions.html), that you can then pass to clang-format using the [-style](-style=file) flag.

**Describe the solution you'd like**
Support passing the configuration files to clang-format.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by admiralAwkbar** [comment]
@ferrarimarco I can probably get around to this one today :)

Thanks for the ping!

**Comment by admiralAwkbar** [comment]
@ferrarimarco also congrats on winning the Euro cup!

**Comment by ferrarimarco** [comment]
Ah! thanks mate :)

I guess the question is what happens if clang-format doesn't find that file.

**Comment by admiralAwkbar** [comment]
@ferrarimarco I will try to find default and have it stored in `TEMPLATES` hopefully it's straightforward...  but we have seen this become fun...

**Comment by ferrarimarco** [comment]
Yes, also because that path navigation algorithm is going to be fun any time you don't have that file in the tree, like we do in this repo for example.

**Comment by ferrarimarco** [comment]
Also, a fun implementation would involve passing the contents of a file as a JSON (apparently clang-format supports that).

Or maybe a symbolic link is easier. :)

**Comment by lum1n0us** [comment]
perhaps 
```
  --fallback-style=<string>  - The name of the predefined style used as a
                               fallback in case clang-format is invoked with
                               -style=file, but can not find the .clang-format
                               file to use.
                               Use -fallback-style=none to skip formatting. 
```

**Comment by morphogencc** [comment]
is there any movement on this issue?  I actually came here to post this same issue but saw this was already in progress!

**Comment by ImpossibleShape** [comment]
I would like to know progress as well

**Comment by Aurumaker72** [comment]
Same here
